lib/sacastats_web/views/character_view.ex

```diff
@@ -62,6 +62,27 @@ defmodule SacaStatsWeb.CharacterView do
     |> Enum.join(" ")
   end
 
+  def build_kills_by_weapon(assigns, %Session{} = session) do
+    weapon_kill_counts =
+      for %Death{} = death <- session.deaths, reduce: %{} do
+        weapon_kill_counts ->
+          if is_nil(SacaStats.weapons()[death.attacker_weapon_id]) do
```

Comment:
Should probably add a check if killer id = character id

Reply:
yes good catch, I'll add that

Reply:
Just pushed, let me know if it looks good